build/Android/XplatCppSdk/src/main/java/com/microsoft/xplatcppsdk/HttpRequest.java

```diff
@@ -30,47 +31,79 @@
     }
 
     public HttpRequest() {
-        requestBuilder = new Request.Builder();
-        requestCompleted = false;
+        this.requestBuilder = new Request.Builder();
     }
 
-    public static HttpRequest createRequest() {
-        return new HttpRequest();
+    public void setMethod(String method) {
+        this.method = method;
     }
 
-    public void setUrl(String url) {
-        this.requestBuilder = this.requestBuilder.url(url);
+    public boolean setUrl(String url) {
+        try
+        {
+            this.requestBuilder = this.requestBuilder.url(url);
+        }
+        catch (Exception e)
+        {
+            return false;
+        }
+        return true;
     }
 
     public void setHeader(String name, String value) {
-        this.requestBuilder = requestBuilder.addHeader(name, value);
+        try
+        {
+            if(name.equalsIgnoreCase("Content-Type"))

```

Comment:
are we planning on using some other content-type except JSON?

Reply:
This is bridge file and it is usually transparently working. This should be decide in PlayFab side.